csr_check_test.go

```diff
@@ -2296,3 +2873,48 @@ func assertNoChange(t *testing.T, a, b []string, f func(*testing.T)) {
 			"new a = %v, new b = %v", aCopy, bCopy, a, b)
 	}
 }
+
+func parseCert(t *testing.T, cert string) *x509.Certificate {
+	block, _ := pem.Decode([]byte(cert))
+	parsedCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("Passed invalid Certificate: %v ", err)
+	}
+	return parsedCert
+}
+
+func parseCR(t *testing.T, csr string) *x509.CertificateRequest {
+	block, _ := pem.Decode([]byte(csr))
+	parsedCR, err := x509.ParseCertificateRequest(block.Bytes)
+	if err != nil {
+		t.Fatalf("Passed invalid Certificate: %v ", err)
+	}
+	return parsedCR
+}
+
+func fakeResponder(t *testing.T, laddr, cert, key string) (server net.Listener) {
+	crt, err := tls.X509KeyPair([]byte(cert), []byte(key))
+	if err != nil {
+		t.Fatalf("Fail to parse key pair: %s", err.Error())
+	}
+
+	server, err = tls.Listen("tcp", laddr, &tls.Config{
+		GetCertificate: func(_ *tls.ClientHelloInfo) (*tls.Certificate, error) { return &crt, nil },
+	})
+	if err != nil {
+		t.Fatalf("Fail to establish TCP listener: %s", err.Error())
+	}
+
+	return
+}
+
+func respond(server net.Listener) {
```

Comment:
Am I right in thinking `Accept` will return an error if the listener is shutdown?

Reply:
It will. I just needed a consumer, so ignoring all the errors worked for me in such a scenario. One pass is guaranteed to be a success, so the code accepts and responds only when it's needed.